common/presenters/population-table/locations-to-population-table-component.js

```diff
@@ -0,0 +1,70 @@
+const { parseISO, isValid } = require('date-fns')
+
+const tablePresenters = require('../table')
+
+const { dayConfig, establishmentConfig } = require('./population-table-config')
+
+function locationsToPopulationTable({
+  isSortable = false,
+  query,
+  focusDate,
+  startDate,
+} = {}) {
+  if (!isValid(startDate)) {
```

Comment:
I was going to suggest that some of this is already handled within [filters](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/eb2ca24fd190a1807e6280acc49f70d1fb076444/config/nunjucks/filters.js#L53) but I can see that you manipulate the date after this.

You could potentially call this:

```javascript
function locationsToPopulationTable({
  ...
  startDate = new Date(),
} = {}) {
  const tableConfig = [establishmentConfig({})]

  for (i = 0; i < 7; i++) {
    const parsedDate = isDate(startDate) ? startDate : parseISO(startDate)
    const dateWithOffset = addDays(parsedDate, i)
    const day = dayConfig({
      focusDate,
      date: dateWithOffset,
      baseOffset: i,
    })

    tableConfig.push(day)
  }

  return tableConfig
}
```

**Disclaimer**: this is just an example of how I think it could be simplified, but I understand tests, etc have been written so I'll leave it up to you what's worth taking from this. It could also be left for a future change if you ever come back to this piece of code.

Reply:
With the addition of looping over day config,  the code is a lot simpler now.

I'll come back to the idea of reusing the `filters` once this has been properly integrated.